File: design-patterns/java/state/ItemSelectedState.java
public class ItemSelectedState implements MachineState {
    @Override
    public void selectItem(VendingMachine context, String itemCode) {
        System.out.println("Item already selected: " + context.getSelectedItem());
    }

    @Override
    public void insertCoin(VendingMachine context, double amount) {
        System.out.println("Inserted $" + amount + " for item: " + context.getSelectedItem());
        context.setInsertedAmount(amount);
        context.setState(new HasMoneyState());
    }

    @Override
    public void dispenseItem(VendingMachine context) {
        System.out.println("Insert coin before dispensing.");
    }
}

File: design-patterns/csharp/state/ItemSelectedState.cs
using System;

namespace State
{
    public class ItemSelectedState : IMachineState
    {
        public void SelectItem(VendingMachine context, string itemCode)
        {
            Console.WriteLine($"Item already selected: {context.GetSelectedItem()}");
        }

        public void InsertCoin(VendingMachine context, double amount)
        {
            Console.WriteLine($"Inserted ${amount} for item: {context.GetSelectedItem()}");
            context.SetInsertedAmount(amount);
            context.SetState(new HasMoneyState());
        }

        public void DispenseItem(VendingMachine context)
        {
            Console.WriteLine("Insert coin before dispensing.");
        }
    }
} 